docs/coding.md

```diff
@@ -0,0 +1,75 @@
+# Coding Conventions
+
+Please follow coding conventions and guidelines described in the following documents:
+
+* [Go proverbs](https://go-proverbs.github.io/) - highly recommended read
+* [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Effective Go](https://golang.org/doc/effective_go.html)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+Here's a list of some more specific conventions that are often followed in
+the code and will be pointed out in the review process:
+
+## General
+
+* Keep variable names short for variables that are local to the function.
+* Do not export a function or variable name outside the package until you
+  have an external consumer for it.
+* Do not use named return values in function definitions. Use only the type.
+  Exception: defer()'d functions.
+
+### Imports
+
+We use the following convention for specifying imports:
+
+```
+<import standard library packages>
+
+<import third-party packages>
+
+<import ceph-csi packages>
+```
+
+Example:
+
+```go
+import (
+ "os"
+ "path"
+ "strings"
+ "time"
+
+ "github.com/pborman/uuid"
+ "github.com/pkg/errors"
+
+ "github.com/ceph/ceph-csi/pkg/util"
+)
+```
+
+### Error Handling
+
+* Use variable name `err` to denote error variable during a function call.
+* Reuse the previously declared `err` variable as long as it is in scope.
+  For example, do not use `errWrite` or `errRead`.
+* Do not panic() for errors that can be bubbled up back to user. Use panic()
+  only for fatal errors which shouldn't occur.
+* Do not ignore errors using `_` variable unless you know what you're doing.
+* Error strings should not start with a capital letter.
```

Comment:
Linter catches this, as a result I would remove this from this list. Less rules to remember for the contributor.